src/common/net.cc

```diff
@@ -52,12 +52,71 @@ Ipv4::any() {
 
 std::string
 Ipv4::toString() const {
-    static constexpr size_t MaxSize = sizeof("255") * 4 + 3 + 1; /* 4 * 255 + 3 * dot + \0 */
+    
+    // Use the built-in ipv4 string length from arpa/inet.h
+    char buff[INET_ADDRSTRLEN+1];
+    
+    in_addr_t addr;
+    toNetwork(&addr);
+    
+    // Convert the network format address into display format
+    inet_ntop(AF_INET, &addr, buff, INET_ADDRSTRLEN);
+    
+    return std::string(buff);
+}
 
-    char buff[MaxSize];
-    snprintf(buff, MaxSize, "%d.%d.%d.%d", a, b, c, d);
+void Ipv4::toNetwork(in_addr_t *addr) const {
+    // Bitshift the bytes into an in_addr_t (a single 32bit unsigned int);
+    *addr = (uint32_t)(d<<24) | (uint32_t)(c<<16) | (uint32_t)(b<<8) | (uint32_t)a;
```

Comment:
Will this behavior be correct on a big-endian cpu?

Reply:
That's a good question, the short answer is that I'm not certain. I'll double check.

Reply:
I haven't gotten around to validating if this is actually a problem as I don't have a big-endian system at home, however something like:
`*addr = htonl( (uint32_t)(a<<24) | (uint32_t)(b<<16) | (uint32_t)(c<<8) | (uint32_t)d );`
Should ensure that the resulting number is compatible across all architectures by making it the compiler's responsibility (notice the change in the ordering of the bytes). 
If that's an acceptable solution I'll put it in a commit, otherwise I'll see if I can come up with a cleaner way.